docs/sources/flow/reference/components/pyroscope.scrape.md

```diff
@@ -32,45 +55,104 @@ pyroscope.scrape "LABEL" {
-`bearer_token`             | `secret`   | Bearer token to authenticate with. | | no
-`bearer_token_file`        | `string`   | File containing a bearer token to authenticate with. | | no
-`proxy_url`                | `string`   | HTTP proxy to proxy requests through. | | no
-`follow_redirects`         | `bool`     | Whether redirects returned by the server should be followed. | `true` | no
-`enable_http2`             | `bool`     | Whether HTTP2 is supported for requests. | `true` | no
-
- At most one of the following can be provided:
+Name                | Type                     | Description                                                  | Default        | Required
+------------------- | ------------------------ | ------------------------------------------------------------ | -------------- | --------
+`targets`           | `list(map(string))`      | List of targets to scrape.                                   |                | yes
+`forward_to`        | `list(ProfilesReceiver)` | List of receivers to send scraped profiles to.               |                | yes
+`job_name`          | `string`                 | The job name to override the job label with.                 | component name | no
+`params`            | `map(list(string))`      | A set of query parameters with which the target is scraped.  |                | no
+`scrape_interval`   | `duration`               | How frequently to scrape the targets of this scrape config.  | `"15s"`        | no
+`scrape_timeout`    | `duration`               | The timeout for scraping targets of this config.             | `"18s"`        | no
+`scheme`            | `string`                 | The URL scheme with which to fetch metrics from targets.     | `"http"`       | no
+`bearer_token`      | `secret`                 | Bearer token to authenticate with.                           |                | no
+`bearer_token_file` | `string`                 | File containing a bearer token to authenticate with.         |                | no
+`proxy_url`         | `string`                 | HTTP proxy to proxy requests through.                        |                | no
+`follow_redirects`  | `bool`                   | Whether redirects returned by the server should be followed. | `true`         | no
+`enable_http2`      | `bool`                   | Whether HTTP2 is supported for requests.                     | `true`         | no
+
+At most one of the following authentication mechanisms can be provided:
  - [`bearer_token` argument](#arguments).
  - [`bearer_token_file` argument](#arguments).
  - [`basic_auth` block][basic_auth].
  - [`authorization` block][authorization].
  - [`oauth2` block][oauth2].
 
- [arguments]: #arguments
+[arguments]: #arguments
+
+#### `job_name` argument
+
+`job_name` defaults to the component's unique identifier.
+
+For example, the `job_name` of `pyroscope.scrape "local" { ... }` will be `"pyroscope.scrape/local"`.
+
+#### `targets` argument
```

Comment:
It's not normal to add a section for an argument inside the `## Arguments` section. However, I think it really helps here because there's lots to discuss. I added the sections using `####` and not `###` so that they don't stand out too much and so that they are not listed in the ToC:
![Screenshot 2023-11-24 at 13 28 50](https://github.com/grafana/agent/assets/5834788/269f6519-c83a-4e38-91dc-38c6892bedf2)